azext_edge/edge/providers/support/base.py

```diff
@@ -164,7 +164,7 @@ def process_statefulset(
     return_namespaces: bool = False,
     field_selector: Optional[str] = None,
     label_selector: Optional[str] = None,
-) -> List[dict]:
+) -> Union[Tuple, List[dict]]:
```

Comment:
when using Tuple from typing, might as well provide types for the generic i.e. `Union[Tuple[List[dict], dict], List[dict]]`